sqlmesh/core/selector.py

```diff
@@ -108,35 +178,29 @@ def expand_model_selections(self, model_selections: t.Iterable[str]) -> t.Set[st
         """
         result: t.Set[str] = set()
 
-        def _add_model(model_name: str, include_upstream: bool, include_downstream: bool) -> None:
-            result.add(model_name)
-            if include_upstream:
-                result.update(self._dag.upstream(model_name))
-            if include_downstream:
-                result.update(self._dag.downstream(model_name))
-
         for selection in model_selections:
             if not selection:
                 continue
 
-            include_upstream = False
-            include_downstream = False
-            if selection[0] == "+":
-                selection = selection[1:]
-                include_upstream = True
-            if selection[-1] == "+":
-                selection = selection[:-1]
-                include_downstream = True
+            (
+                selection,
+                include_upstream,
+                include_downstream,
+            ) = self._get_value_and_dependency_inclusion(selection)
 
             if "*" in selection:
                 for model in self._models.values():
-                    if fnmatch.fnmatch(model.name, selection):
-                        _add_model(model.fqn, include_upstream, include_downstream)
+                    if fnmatch.fnmatchcase(model.name, selection):
```

Comment:
@izeigerman I think you wanted to use `fnmatchcase` here instead of `fnmatch`. `fnmatch` determines if the matching is case-sensitive based on the OS which feels irrelevant within this context. Therefore I think always matching on case makes more sense here.